lib/receiver.rb

```diff
@@ -11,12 +10,22 @@ def input
     @input_function.gets.chomp
   end
 
-  def get_player_move(board, mark)
+  def get_player_move(display, board, mark)
     move = input
 
     until @validator.valid_move?(board, move, @converter)
+      display.output(INPUT_ERROR)
       move = input
     end
     @converter.numberfy(move)
   end
+
+  def get_player_two_choice(display)
+    choice = input
+    until @validator.valid_player2?(choice)
+      display.output(INPUT_ERROR)
+      choice = input
+    end
+    choice
+  end
 end
```

Comment:
If you look closely, get_player_move is very similar to get_player_two_choice.  Could you make this one generic method of get_player_input (regardless of whether related to the move or the game choice)?  This is where having different validator's makes sense.  You can pass in a different validator depending on if you want a move or a choice.  The different validators have the same method (could be called valid?).  The validators, depending on which one, could be initialized with the board.  In terms of the converter, I'll let you figure out how to weave it in.